### PR TITLE
 Convert to use ed25519 keys / multihash.

### LINKED ISSUE #24: Use veres-one-context external package
Refactor to use the Veres One `@context` provided by [`veres-one-context`](https://github.com/veres-one/veres-one-context) instead of the context included in this repo, once https://github.com/veres-one/veres-one-context/issues/2 is implemented.

### PR DESCRIPTION
**Pull request by dmitrizagidulin**
Based on Manu's work on did-veres-one (https://github.com/veres-one/did-veres-one/pull/6).

Closes #24 and #25.

(Depends on PR https://github.com/veres-one/veres-one-context/pull/3 ).

### CODE REVIEW

**Comment by dmitrizagidulin** [comment]
Ready to review.

File: lib/methods/veres-one/veres-did-doc.js [hunk 1]
```diff
@@ -100,23 +104,31 @@ class VeresOneDidDoc {
...
    */
-  generateId({authKey, didType = 'nym', env = 'dev'}) {
+  generateId({keyPair, didType = 'nym', env = 'dev'}) {
     if(didType === 'uuid') {
       const prefix = (env === 'live') ? 'did:v1:' : 'did:v1:test:';
       return (prefix + 'uuid:' + uuid()).replace(/-/g, '');
     }
 
+    if(!keyPair) {
+      throw new Error('Cannot generate a cryptonym DID without a key')
```
Comment (mattcollier) [inline lib/methods/veres-one/veres-did-doc.js]: missing semi.  There are a number of linting violations in the files in this repo.  Please lint with the latest rules.

File: lib/methods/veres-one/veres-did-doc.js [hunk 2]
```diff
@@ -100,23 +104,31 @@ class VeresOneDidDoc {
...
+    }
+
     // didType === 'nym'
     return this.createCryptonymDid({
-      publicKey: authKey.publicKey,
-      encoding: typeof authKey.publicKey === 'string' ?
-        'base58' : 'forge',
+      publicKey: keyPair.publicKey,
+      encoding: typeof keyPair.publicKey === 'string' ?
+        'ed25519' : 'forge',
```
Comment (mattcollier) [inline lib/methods/veres-one/veres-did-doc.js]: one liner.

**Review by mattcollier — CHANGES_REQUESTED** [review]
I thought the aim was to have a did-veres-one dependency and use that code the generate the DID.  Is that correct?

File: lib/methods/veres-one/veres-did-doc.js (hunk 1 above)
Comment (dmitrizagidulin) [reply lib/methods/veres-one/veres-did-doc.js]: Thanks! Re-linted and fixed.

**Comment by dmitrizagidulin** [comment]
> I thought the aim was to have a did-veres-one dependency and use that code the generate the DID. Is that correct?

The plan is to eventually extract Veres-related code out of did-io into did-veres-one, but it is not a priority. All downstream libs should be using `did-io` meanwhile, anyway.

File: lib/methods/veres-one/veres-did-doc.js (hunk 2 above)
Comment (dmitrizagidulin) [reply lib/methods/veres-one/veres-did-doc.js]: Fixed, thanks.

**Comment by mattcollier** [comment]
@dmitrizagidulin one of the tests is failing on node 8: https://travis-ci.org/digitalbazaar/did-io/jobs/435714189

**Comment by dmitrizagidulin** [comment]
@mattcollier:

> one of the tests is failing on node 8: https://travis-ci.org/digitalbazaar/did-io/jobs/435714189

Yeah, that one's weird, it passes like 3/4 of the time, so something is off either with the algorithm or with the test. I need to check with @msporny or @dlongley

**Comment by dmitrizagidulin** [comment]
Ok, I extracted fingerprint generating code to `LDKeyPair` class, and added a corresponding `verifyFingerprint()` function. The reason the test was failing had to do with the vagaries of base58-encoding. When we compare bytes directly, the fingerprints verify.

File: lib/ld-key-pair.js [hunk 3]
```diff
@@ -207,6 +221,43 @@ class Ed25519KeyPair extends LDKeyPair {
...
+   *
+   * @returns {boolean}
+   */
+  verifyFingerprint(fingerprint) {
+    const forge = this.injector.use('node-forge');
+    const fingerprintBuffer = forge.util.binary.base58.decode(fingerprint);
+    const publicKeyBuffer = forge.util.binary.base58.decode(this.publicKey);
+
+    // slice off the two first multiformat bytes, 0x30 and 0xed
+    return publicKeyBuffer.equals(fingerprintBuffer.slice(2));
```
Comment (mattcollier) [inline lib/ld-key-pair.js]: has the information being discarded here already been validated?  Does it need to be?

**Review by mattcollier — APPROVED** [review]

File: lib/ld-key-pair.js (hunk 3 above)
Comment (dmitrizagidulin) [reply lib/ld-key-pair.js]: Erm, fair point. Now validating.

**Review by dlongley — APPROVED** [review]
LGTM

**Pull request merged**